src/FlapperUniV2.sol

```diff
@@ -150,14 +171,17 @@ contract FlapperUniV2 {
         require(block.timestamp >= zzz + hop, "FlapperUniV2/kicked-too-soon");
         zzz = block.timestamp;
 
+        // Get Dai
+        (uint256 _reserveDai, uint256 _reserveGem) = _getReserves();
         uint256 _wlot = lot / RAY;
-        require(_wlot * RAY == lot, "FlapperUniV2/lot-not-multiple-of-ray");
+        uint256 _total = _getTotalDai(_wlot, _reserveDai);
+        require(_total < _wlot * 220 / 100, "FlapperUniV2/total-insanity");
 
-        // Swap
-        vat.move(msg.sender, address(this), lot);
-        daiJoin.exit(address(this), _wlot);
+        vat.move(msg.sender, address(this), _total * RAY);
+        daiJoin.exit(address(this), _total);
+        //
 
-        (uint256 _reserveDai, uint256 _reserveGem) = _getReserves();
+        // Swap
         uint256 _buy = _getAmountOut(_wlot, _reserveDai, _reserveGem);
         require(_buy >= _wlot * want / (uint256(pip.read()) * RAY / spotter.par()), "FlapperUniV2/insufficient-buy-amount");
```

Comment:
Nit: we might want to do this before the `vat.move`. It seems like better to solve all the requires before doing the actual transfer (thinking that in this case is possible).

Reply:
Done.